packages/mask/src/plugins/Pets/SNSAdaptor/PetDialog.tsx

```diff
@@ -0,0 +1,283 @@
+import {
+    Button,
+    TextField,
+    Typography,
+    Box,
+    DialogContent,
+    Grid,
+    MenuItem,
+    Snackbar,
+    Autocomplete,
+} from '@mui/material'
+import { PluginPetMessages, PluginPetRPC } from '../messages'
+import { InjectedDialog } from '../../../components/shared/InjectedDialog'
+import { initMeta, initCollection } from '../constants'
+import { PreviewBox } from './previewBox'
+import type { PetMetaDB, FilterContract, CollectionNFT } from '../types'
+import { useUser, useNfts } from '../hooks'
+import { useI18N } from '../../../utils'
+import { ShadowRootPopper } from '../../../utils/shadow-root/ShadowRootComponents'
+
+const useStyles = makeStyles()((theme) => ({
+    desBox: {
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+    des: {
+        color: '#7b8192',
+        fontSize: '12px',
+    },
+    input: {
+        margin: theme.spacing(2, 0, 0),
+    },
+    inputOptl: {
+        margin: theme.spacing(4, 0, 0),
+    },
+    inputBorder: {
+        borderRadius: theme.spacing(1),
+        padding: theme.spacing(1),
+    },
+    inputArea: {
+        borderRadius: theme.spacing(1),
+        padding: theme.spacing(2),
+    },
+    menuItem: {
+        width: '100%',
+    },
+    btn: {
+        margin: theme.spacing(8, 0, 4),
+    },
+    thumbnail: {
+        width: 25,
+        height: 25,
+        objectFit: 'cover',
+        margin: theme.spacing(0, 1, 0, 0),
+        display: 'inline-block',
+        borderRadius: 4,
+    },
+    itemFix: {
+        // flex: 1,
+        display: 'flex',
+        alignItems: 'center',
+        width: '100%',
+    },
+    itemTxt: {
+        flex: 1,
+    },
+    prevBox: {
+        margin: theme.spacing(2, 0, 0),
+        border: '1px dashed #ccc',
+        borderRadius: 4,
+        height: 'calc(100% - 16px)',
+        boxSizing: 'border-box',
+        padding: 4,
+    },
+}))
+
+export function PetDialog() {
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), {})
+    const chainId = useChainId()
+    const user = useUser()
+    const nfts = useNfts(user)
+    const [extraData, setExtraData] = useState<CollectionNFT[]>([])
+    const { open, closeDialog } = useRemoteControlledDialog(PluginPetMessages.essayDialogUpdated, () => {})
+
+    const [collection, setCollection] = useState<FilterContract>(initCollection)
+    const [isCollectionsError, setCollectionsError] = useState(false)
+
+    const [metaData, setMetaData] = useState<PetMetaDB>(initMeta)
+    const [isImageError, setImageError] = useState(false)
+    const [isTipShow, setTipShow] = useState(false)
+    const [holderChange, setHolderChange] = useState(true)
+
+    useEffect(() => {
+        if (!open) {
+            setMetaData(initMeta)
+            setCollection(initCollection)
+        }
+    }, [open])
+
+    useEffect(() => {
+        Promise.all(nfts.map((i) => PluginPetRPC.getAssetContract(i.contract, chainId))).then((lists) => {
```

Comment:
This code may run into a race condition. Refactor it to use [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).